gwcs/coordinate_frames.py

```diff
@@ -0,0 +1,402 @@
+        Name for this frame.
+
+    """
+
+    def __init__(self, reference_frame, axes_order=(0,), unit=None, axes_names=None, name=""):
+        super(SpectralFrame, self).__init__(naxes=1, reference_frame=reference_frame,
+                                            axes_order=axes_order,
+                                            axes_names=axes_names,
+                                            unit=unit, name=name)
+
+    def __repr__(self):
+        fmt = "<SpectralFrame(reference_frame={0}, unit={1}, axes_names={2}, axes_order={3}, name={4})>".format(
+            self.reference_frame, self.unit,
+            self.axes_names, self.axes_order, self.name)
+        return fmt
+
+    def world_coordinates(self, value):
+        """
+        Create a SkyCoord object.
+
+        Parameters
+        ----------
+        value : float
+            Coordinate value.
+        """
+        return self.reference_frame.realize_frame(Cartesian1DRepresentation(value * u.Unit(self.unit[0])))
+
+    def transform_to(self, x, other):
+        """
+        Transform from the current reference frame to other.
+
+        Parameters
+        ----------
+        x : float or array-like
+        other : str or `BaseCoordinateFrame` class
+            The frame to transform this coordinate into.
+        """
+        return self.world_coordinates(x).transform_to(other)
+
+
+class CompositeFrame(CoordinateFrame):
+    """
+    Represents one or more frames.
+
+    Parameters
+    ----------
+    frames : list
+        List of frames (TimeFrame, CelestialFrame, SpectralFrame, CoordinateFrame).
+    name : str
+        Name for this frame.
+
+    """
+    def __init__(self, frames, name=""):
+        """
+        naxes, axes_order=(0, 1), reference_frame=None, reference_position=None,
+                 unit=None, axes_names=None, name=None):
+
+
+        """
+        self._frames = frames[:]
+        naxes = sum([frame._naxes for frame in self._frames])
+        unit = []
+        axes_order = []
+        axes_names = []
+        for frame in frames:
+            unit.extend(frame.unit)
+            axes_order.extend(frame.axes_order)
+            axes_names.extend(frame.axes_names)
+        super(CompositeFrame, self).__init__(naxes, axes_order=axes_order,
+                                             unit=unit, axes_names=axes_names,
+                                             name=name)
+
+    @property
+    def frames(self):
+        return self._frames
+
+    def __repr__(self):
+        return repr(self.frames)
+
+    def world_coordinates(self, *args):
+        """
+        Create world coordinates.
+        """
+        if len(args) != self.naxes:
+            raise TypeError("Expected {0} arguments ({1} given)".format(len(args), self.naxes))
+        result = ()
+        n = 0
+        for frame in self.frames:
+            result += (frame.world_coordinates(*args[n : frame.naxes + n]), )
+            n = frame.naxes
+        return result
```

Comment:
This can come later, but we'll probably want some tools to work with this.  Say you have some complicated nested thing like:

```
[CelestialFrame(...), CompositeFrame([SpectralFrame(...), TimeFrame(...)])]
```

it would probably be helpful to have a method that takes an axis number and returns the frame object it's actually in as well as an index into that frame (in the case of multi-axis frames like Celestial).